### PR TITLE
return from ossl_namemap_doall_names() immediately if namemap is NULL

### PR DESCRIPTION
**Pull request by vladak**
When `ossl_namemap_stored()` returns NULL in
https://github.com/openssl/openssl/blob/6c0ecc2bce64cc86948a51f80f832b5e48a9ebea/crypto/evp/evp_fetch.c#L621-L629 then `ossl_namemap_doall_names()` will attempt to dereference it. This trivial change fixes that. I am fixing this because this actually lead to a crash in my environment.

### CODE REVIEW

File: crypto/core_namemap.c [hunk 1]
```diff
@@ -137,6 +137,10 @@ int ossl_namemap_doall_names(const OSSL_NAMEMAP *namemap, int number,
     cbdata.number = number;
     cbdata.found = 0;
 
+    if (namemap == NULL) {
+        return 0;
+    }
```
Comment (mattcaswell) [inline crypto/core_namemap.c:142]: OpenSSL style is to not include `{}` for a single line if.

Comment (vladak) [reply crypto/core_namemap.c]: Ok, fixed.

**Review by mattcaswell — APPROVED** [review]

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by paulidale** [comment]
Merged, thanks for the fix.

**Pull request closed without merging**